src/main/java/com/atlassian/db/replica/api/reason/RouteDecision.java

```diff
@@ -38,6 +38,20 @@ public Reason getReason() {
         return Optional.ofNullable(cause);
     }
 
+    /**
+     * @return True if the database call would always fail on a read replica.
+     */
+    public boolean isWrite() {
+        return reason.isWrite();
+    }
+
+    /**
+     * @return Decision whether current route is Run on Main or Replica
+     */
+    public boolean isRunOnMain() {
```

Comment:
"Current route" is not well defined.
```suggestion
    /**
     * @return true if the accompanying {@link SqlCall} will be run on the main database.
     */
    public boolean willRunOnMain() {
```
`SqlCall` has to be imported or fully-qualified for proper linking.